lib/pbench/test/unit/server/test_indexing_tarballs.py

```diff
@@ -0,0 +1,54 @@
+import json
+import pytest
+
+# from pbench.server.indexing_tarballs import Index
+
+
+class AccessJSONFile:
+    def __init__(self):
+        self._data = None
+
+    def access(self, data_file):
+        try:
+            with open(data_file) as jf:
+                self._data = json.load(jf)
+        except Exception as exc:
+            raise FileNotFoundError(str(exc))
+
+    def get_data(self):
+        date = None
+        try:
+            date = self._data["_source"]
+        except Exception as e:
+            print("date error ", e)
+        return date
```

Comment:
Is this code all gone, now?  (I cannot find it.)

I wanted to point out that catching all exceptions at line 15 and then raising `FileNotFoundError` is probably not a good idea, as the real exception might be that the file is present but contains invalid JSON.  Including the stringization of the original exception as the message for the resulting exception helps, but, if what you want to do is to report the original exception, then you probably don't need to catch it here (it will report itself...).  Alternatively, you can use the `raise FileNotFoundError from exc` idiom, but, again, it's a tad confusing/misleading to be reporting a file-not-found issue if the root cause is bad file contents.  (A more generic exception, like `RuntimeError`, might be a better choice.)

Similarly, the `try` block at line 20 is a bit of a "big hammer".  There are only a few things which could go wrong at line 21 (i.e., `_data` is not a `dict` (e.g., it's `None`), or it doesn't contain the `_source` key), they are things which can be detected locally and easily, and testing for them explicitly would enable more descriptive diagnostics.

However, if this code has been removed, then I suppose it's moot.  :-)

Reply:
I just thought of adding tests at the end and also I was a lot confused about them so will add a different commit for them

Reply:
I will take care of these points and improve that test